File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        int n = nums.size();
        long long totalPairs = ((long long)n * (long long)(n - 1)) / 2, goodPairs = 0;
        unordered_map<int, long long> hash;

        for (int i = 0; i < n; i++) {
            if (hash.find(nums[i] - i) != hash.end()) goodPairs += hash[nums[i] - i];
            hash[nums[i] - i]++;
        }

        return totalPairs - goodPairs;
    }
};

File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.go
func countBadPairs(nums []int) int64 {
    n := len(nums)
    totalPairs, goodPairs := int64((n * (n - 1)) / 2), int64(0)
    hash := make(map[int]int64)

    for i := 0; i < n; i++ {
        if freq, ok := hash[nums[i] - i]; ok {
            goodPairs += freq
        }
        hash[nums[i] - i]++
    }

    return totalPairs - goodPairs
}